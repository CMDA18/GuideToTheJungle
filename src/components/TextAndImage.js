// @flow
import React from 'react'
import styled from 'styled-components'
import {
  textStyles,
  colorPalette
} from '../styles/styleGuide'
import { fromBreakpoint } from '../styles/breakpoints'

type ITextMedia = {
  title?: string,
  text: string
}

type Props = {
  imageAlignment: string,
  text: Array<ITextMedia>,
  image: string,
  mediaCaption: string
}

const TextAndImage = (props: Props) => {
  return (
    <Container imageAlignment={props.imageAlignment}>
      <TextContainer imageAlignment={props.imageAlignment}>
        {(props.text || []).map((item, index) => (
          <Wrapper key={index} >
            <Title>{item.title}</Title>
            <Text dangerouslySetInnerHTML={{__html: `${item.text}`}} />
          </Wrapper>
        ))}
      </TextContainer>
      <TriangleWrapper>
      </TriangleWrapper>
      <ImageContainer imageAlignment={props.imageAlignment}>
        <Media
          backgroundImage={props.image ? props.image : props.image}
        />
        <Caption>{props.mediaCaption}</Caption>
      </ImageContainer>

    </Container>
  )
}

export default TextAndImage

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 40px;
  align-items: flex-start;
  flex-direction: row;
`

const TextContainer = styled.div`
  ${fromBreakpoint.lg`
      width: 600px;
      flex-shrink: 0;

    ${props =>
    props.imageAlignment === 'left'
      ? `
        order: 2;
        margin-left: 50px;
      `
      : `
        order: 1;
        margin-right: 50px;
    `}
  `};
`

const Wrapper = styled.div`
  margin-bottom: 50px;
`

const TriangleWrapper = styled.div`
  position: absolute;
  right: 70px;
  top: 350px;
  display: none;
`

const Title = styled.h1`
  ${textStyles.title};
`

const Text = styled.p`
  margin-top: 0;
  transition: color 0.3s ease-in-out;
  color: ${colorPalette.blue};
  ${textStyles.general};
  margin-bottom: -20px;
`

const ImageContainer = styled.div`
  max-width: 400px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  align-self: center;
  z-index: 2;
  margin-top: -160px;
  
  :before {
    content : '';
    width: 0;
    height: 0;
    border-width: 165px 113.5px 0 113.5px;
    border-color: ${colorPalette.orange} transparent transparent transparent;
    border-style: solid;
    position: relative;
    z-index: -1;
    top: 289px;
    right: 73px;
    transform: rotate(47deg) scale(2);
  }

  ${fromBreakpoint.lg`
  order: ${props =>
    props.imageAlignment === 'left'
      ? `
        1;
      `
      : `
        2;
        `}
  `};
`

const Media = styled.div`
  background-image: url(${props =>
    props.backgroundImage ? props.backgroundImage : ''});
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  width: 600px;
  height: 300px;
`

const Caption = styled.p`
  display: block;
  font-size: 16px;
  line-height: 24px;
  margin-top: 30px;
  color: ${colorPalette.blue};
`
