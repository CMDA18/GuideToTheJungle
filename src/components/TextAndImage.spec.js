import React from 'react'
import { shallow } from 'enzyme'
import TextAndImage from './TextAndImage'

it('renders without crashing', () => {
  expect(shallow(<TextAndImage />)).toMatchSnapshot()
})
