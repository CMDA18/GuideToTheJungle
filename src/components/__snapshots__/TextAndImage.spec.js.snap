// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextAndImage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TextAndImage__TextContainer
          imageAlignment={undefined}
        />,
        <TextAndImage__TriangleWrapper />,
        <TextAndImage__ImageContainer
          imageAlignment={undefined}
        >
          <TextAndImage__Media
            backgroundImage={undefined}
          />
          <TextAndImage__Caption />
        </TextAndImage__ImageContainer>,
      ],
      "imageAlignment": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
          "imageAlignment": undefined,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextAndImage__Media
              backgroundImage={undefined}
            />,
            <TextAndImage__Caption />,
          ],
          "imageAlignment": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "backgroundImage": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextAndImage__TextContainer
            imageAlignment={undefined}
          />,
          <TextAndImage__TriangleWrapper />,
          <TextAndImage__ImageContainer
            imageAlignment={undefined}
          >
            <TextAndImage__Media
              backgroundImage={undefined}
            />
            <TextAndImage__Caption />
          </TextAndImage__ImageContainer>,
        ],
        "imageAlignment": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
            "imageAlignment": undefined,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TextAndImage__Media
                backgroundImage={undefined}
              />,
              <TextAndImage__Caption />,
            ],
            "imageAlignment": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "backgroundImage": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
